Payments/merge charge with capture (#4098)
mirumee/saleor: docs/guides/payments.rst

.. note::

    All the below functions receive ``payment_information`` as a dataclass: ``PaymentData`` and ``config`` as a dataclass: ``ConfigData``. 

>     Functions should return a response as a dataclass: ``GatewayResponse``. 

Reviewer: ```suggestion
    Those functions should return a response as a dataclass: ``GatewayResponse``. 
```

+-------------------------+-----------------+-----------------------------------------------------------------------------+
| ``payment_information`` | ``PaymentData`` | Payment information, containing the token, amount, currency and billing.    |
+-------------------------+-----------------+-----------------------------------------------------------------------------+

> | ``config``              | ``ConfigData``  | Configuration of the payment gateway.                                               |

Reviewer 2: nit: isn't last pipe (`|`) too far?

Reviewer: Looks like it

<img width="1054" alt="Screenshot 2019-05-14 at 12 21 02" src="https://user-images.githubusercontent.com/6186720/57690803-c8c84d00-7642-11e9-84b9-29f52e326158.png">

| phone            | ``str`` |
+------------------+---------+

ConfigData
""""""""""

+---------------------+-----------+---------------------------------------------------------------------------------------------------------+
| name                | type      | description                                                                                             |
+---------------------+-----------+---------------------------------------------------------------------------------------------------------+
| auto_capture        | ``bool``  | Define if gateway should also capture funds from the card. If false, payment should be only authorized  |
+---------------------+-----------+---------------------------------------------------------------------------------------------------------+

> | template_path       | ``str``   | Should specify a path to a template that will be rendered for the checkout.                             |

Reviewer 2: ```suggestion
| template_path       | ``str``   | Path to a template that will be rendered for the checkout.                             |
```

.. code-block:: python

    PAYMENT_GATEWAYS = {

>         'braintree': {

Author: My bad. Thanks

- ``connection_params``
    List of parameters used for connecting to the payment's gateway.

- ``auto_capture``

>     Define if the gateway should also capture funds from the card. When auto_capture is set to False, funds will be blocked but manual capture will be required.

Reviewer: ```suggestion
    Define if the gateway should also capture funds from the card. When ``auto_capture`` is set to ``False``, funds will be blocked by the customer's bank for a 7-days period, a manual capture will be required.
```
